lidiabrowser/sync/populate.py

```diff
@@ -17,11 +20,49 @@
     Publication,
     TermGroup,
 )
+from sync.zoteroutils import get_attachment_url, get_attachment_id_from_url
+
 
 logger = logging.getLogger(__name__)
 
 
 LIDIAPREFIX = "~~~~LIDIA~~~~"
+LEXICON_URLS = {}
+
+
+def fetch_lexicon_data():
+    url = "https://github.com/CentreForDigitalHumanities/lidia-zotero/raw/main/vocabulary/lexicon.xlsx"
+    filename = "../data/lexicon.xlsx"
+
+    if os.path.isfile(filename) and os.path.getsize(filename) > 0:
+        logger.info("Lexicon spreadsheet already downloaded.")
+        return
+
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    try:
+        with urllib.request.urlopen(url) as response:
+            with open(filename, 'wb') as out_file:
```

Comment:
Instead of nesting, you can also write the two with a comma in between:
https://docs.python.org/3.9/reference/compound_stmts.html#with